flang/include/flang/Optimizer/Builder/Runtime/Command.h

```diff
@@ -0,0 +1,29 @@
+//===-- Command.h - generate assignment runtime API calls --------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef FORTRAN_OPTIMIZER_BUILDER_RUNTIME_COMMAND_H
+#define FORTRAN_OPTIMIZER_BUILDER_RUNTIME_COMMAND_H
+
+namespace mlir {
+class Value;
+class Location;
+} // namespace mlir
+
+namespace fir {
+class FirOpBuilder;
+}
+
+namespace fir::runtime {
+
+/// Generate a call to the `ArgumentCount` runtime.
+/// This calls the simple runtime entry point that then calls into the more
+/// complex function for calculating the count of command arguments.
```

Comment:
Done. I used "ArgumentCount" in the comment instead of "COMMAND_ARGUMENT_COUNT" since that's what the runtime routine function is actually called.

Reply:
I see there is a precedent of sorts, but I don't agree that it is a good one. If someone new to the code is trying to figure out where the FOOBAR intrinsic from the Fortran standard is implemented, it is more convenient to have the comment use the name FOOBAR rather than ``RazzleDazzle`` because FOOBAR was implemented under some other identifier name.

Reply:
I have also noticed lack of consistency in places and like Eric I am in favor of using names from the spec. `ArgumentCount` is an implementation detail. This is important and relevant, but we should focus on "what" is implemented first (e.g. `COMMAND_ARGUMENT_COUNT`).

> We probably need a larger discussion anyway.
+1

Reply:
> I see there is a precedent of sorts

Yes, the precedents most likely come from me. I actually wrote this as a guideline in https://github.com/flang-compiler/f18-llvm-project/blob/fir-dev/flang/docs/tutorials/addingIntrinsics.md#implementation-3 (see "_If the intrinsic has a runtime implementation, the actual binding should be done in another helper function called gen{Runtime function name}_")

My initial logic was to have functions in `Builder/Runtime/xxx.cpp` called after the runtime names since they are merely runtime call generator wrappers that would be generated automatically if we ever moved from a table gen description of the runtime (or something some kind of description format allowing such compiler function generation).

I kept the intrinsic names in lower/IntrinsicCall.cpp functions (the guideline tells that _"Lowering of intrinsics is driven by the IntrinsicLibrary class. Its member function gen{intrinsic name} lowers "intrinsic name"."_).

The problem is that for some intrinsics, they are different runtime entry points depending on the arguments type/kind/number (e.g. `CSHIFT` has a `RTNAME(CshiftVector)` and a `RTNAME(Cshift)` version), and so far, `IntrinsicCall.cpp` is the one dispatching to the right version according to the needs, and the `Builder/Runtime/xxx.cpp` are really simple wrappers (that dispatch according to the argument kinds, but not much more). The rational for this split is that I had in mind Runtime/xxx.cpp could be automatically generated at some point from the runtime signatures. Also, lowering can decide to generate some easy cases as inlined code, and to only call the runtime for the more complex cases. Hence I stayed away from doing more advanced dispatching.

So if we change the name logic, we will also need to move all the dispatching jobs to the `Builder/Runtime/xxx.cpp`. 
and update addingIntrinsics.md guideline.

Reply:
To be clear, I wasn't proposing changing the runtime or the names of functions.

I was proposing that the doxygen comments reflect the terms used in the Fortran standard rather than ad hoc identifier names from the runtime routines.

Reply:
> To be clear, I wasn't proposing changing the runtime or the names of functions.

Ah, never mind then, I over interpreted your comment. It makes sense to me to use the intrinsic name in the comments.

Reply:
Changed it to COMMAND_ARGUMENT_COUNT as suggested. Makes more sense. 'ArgumentCount' will always show in the source file if someone searched for it, whereas COMMAND_ARGUMENT_COUNT wouldn't. Thanks for the help.